api/cpp/cmake/SlintConfig.cmake.in

```diff
@@ -14,8 +14,41 @@ endif()
 add_library(@slint_cpp_impl@ @cmake_lib_type@ IMPORTED)
 set_target_properties(@slint_cpp_impl@ PROPERTIES @SLINT_LIB_PROPERTIES@)
 
-set(SLINT_COMPILER @SLINT_COMPILER@ CACHE FILEPATH "Path to the slint-compiler executable")
+function(_slint_download_compiler_and_cache)
+    set(SLINT_GITHUB_RELEASE @SLINT_VERSION@ CACHE STRING "GitHub Release to use for Slint Binary Artifact Downloads")
+
+    if (CMAKE_HOST_WIN32)
+        set(compiler_exe_suffix ".exe")
+        set(platform_suffix "windows")
+    else()
+        set(platform_suffix "${CMAKE_HOST_SYSTEM_NAME}-${CMAKE_HOST_SYSTEM_PROCESSOR}")
+    endif()
+
+    set(prebuilt_archive_filename "slint-compiler-${platform_suffix}.tar.gz")
+    set(download_url "https://github.com/slint-ui/slint/releases/download/${SLINT_GITHUB_RELEASE}/${prebuilt_archive_filename}")
+    set(download_directory "${CMAKE_BINARY_DIR}/slint-prebuilt")
+    message(STATUS "Downloading pre-built Slint compiler binary from ${download_url}")
+    file(DOWNLOAD "${download_url}" "${download_directory}/${prebuilt_archive_filename}" STATUS download_status)
+    list(GET download_status 0 download_code)
+    if (NOT download_code EQUAL 0)
+        list(GET download_status 1 download_message)
+        message(FATAL_ERROR "Download of Slint compiler binary package failed: ${download_message}")
```

Comment:
should we perhaps have a link that explain how to get the slint-compiler?  (`cargo install --git ... --tag ...  slint-compiler`?)

Reply:
I was wondering the same but I'm rather unsure: we can recommend to run cargo install and use find_program to search in PATH, but then we also need to implement a full version check to ensure it's correct to avoid mixing Slint 1.9 with the 1.8 compiler. And when the user upgrades they have to re-run cargo install. This all feels out of scope of this PR, that's why I skipped it.

I'd rather learn from this and make it work in all the cases we support (Mac, Windows, Linux, with respective architectures we know of)